spieleplattform-backend/src/main/kotlin/com/example/spieleplattformbackend/game/GameService.kt

```diff
@@ -223,5 +223,13 @@ class GameService(
         return gameRepository.findGamesByGameConsolesContains(gameConsole)
     }
 
+    fun getGamesByNameSearch(nameSearch: String): Iterable<Game> {
+        return gameRepository.findGamesByNameContainsIgnoreCase(nameSearch)
+    }
 
+    fun getGamesByConsoleIdAndNameSearch(consoleId: Int, nameSearch: String): Iterable<Game> {
+        val gameConsole =
+            gameConsoleRepository.findGameConsoleById(consoleId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
```

Comment:
Dein Service Layer, sollte eher keine Http Exceptions werfen. Diese sollten dem Http/Controller-Layer vorbehalten sein. Ein Grund ist auch, dass du deinen Service theoretisch ja auch von einer anderen Stelle in deine Applikation ansprechen willst. Du könntest hier eine andere Exception werfen diese dann im Controller fangen und dort alse ResponseStatusException rethrowen, oder du gibst hier null zurück (also Leere Liste, wenn es die Konsole gibt, aber keine Spiele dazu und null, wenn es die Konsole nicht gibt) und wirfst dann auch wieder die ResponseStatusException im Controller bei null.